
# About

I am currently a bioinformatician and software developer at the [Center for Statistical Genetics](https://sph.umich.edu/csg/), University of Michigan.

# Projects

## Maintainer

Projects on which I am currently a maintainer:

<table>
<thead>
  <tr>
    <th align="left" width="195px">Repo</th>
    <th align="left" width="125px">Contributions</th>
    <th align="left">Description</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td nowrap><a href="https://github.com/statgen/LDServer">LDServer</a></td>
    <td nowrap><a href="https://github.com/statgen/LDServer/commits?author=welchr">Commits</a> • <a href="https://github.com/statgen/LDServer/pulls?q=is%3Apr+author%3Awelchr+">PRs</a></td>
    <td>Serves linkage disequilibrium (LD) data over a REST API. Used to calculate LD on-the-fly for the new interactive <a href="https://github.com/statgen/locuszoom">LocusZoom.js</a> viewer, and score statistics/covariance matrices for use with <a href="https://github.com/statgen/raremetal.js">raremetal.js</a>.</td>
  </tr>
  <tr>
    <td nowrap><a href="https://github.com/statgen/raremetal.js">raremetal.js</a></td>
    <td nowrap><a href="https://github.com/statgen/raremetal.js/commits?author=welchr">Commits</a> • <a href="https://github.com/statgen/raremetal.js/pulls?q=is%3Apr+author%3Awelchr+">PRs</a></td>
    <td>JS package for calculating rare variant aggregation tests. Can calculate in the browser or server-side with node.js. Supports burden, VT, SKAT, and SKAT-O.</td>
  </tr>
  <tr>
    <td nowrap><a href="https://github.com/statgen/locuszoom-api">LocusZoom API</a></td>
    <td nowrap><a href="https://github.com/statgen/locuszoom-api/commits?author=welchr">Commits</a> • <a href="https://github.com/statgen/locuszoom-api/pulls?q=is%3Apr+author%3Awelchr+">PRs</a></td>
    <td>Backend REST API service for <a href="https://github.com/statgen/locuszoom">LocusZoom.js</a>. Serves information on genes, recombination rates, EBI GWAS catalog variants, genomic interval annotations, dbSNP metadata, and more.</td>
  </tr>
</tbody>
</table>

## Contributor

Projects to which I have made a contribution (or at least, attempted to make one):

<table>
<thead>
  <tr>
    <th align="left" width="195px">Repo</th>
    <th align="left" width="125px">Contributions</th>
    <th align="left">Description</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td nowrap><a href="https://github.com/lin-lab/apex">APEX</a></td>
    <td nowrap><a href="https://github.com/lin-lab/apex/commits?author=welchr">Commits</a> • <a href="https://github.com/lin-lab/apex/pulls?q=is%3Apr+author%3Awelchr+">PRs</a></td>
    <td>Toolkit for analysis of molecular quantitative trait loci (xQTLs), such as mRNA expression or methylation levels. Fast and efficient storage of single study summary statisics and covariance matrices.</td>
  </tr>
  <tr>
    <td nowrap><a href="https://github.com/statgen/RAREMETAL">RAREMETAL</a></td>
    <td nowrap><a href="https://github.com/statgen/RAREMETAL/commits?author=welchr">Commits</a> • <a href="https://github.com/statgen/RAREMETAL/pulls?q=is%3Apr+author%3Awelchr+">PRs</a></td>
    <td>Meta-analysis of rare variant burden tests; sharing of summary statistics and covariance matrices.</td>
  </tr>
  <tr>
    <td nowrap><a href="https://github.com/statgen/emeraLD">emeraLD</a></td>
    <td nowrap><a href="https://github.com/statgen/emeraLD/commits?author=welchr">Commits</a> • <a href="https://github.com/statgen/emeraLD/pulls?q=is%3Apr+author%3Awelchr+">PRs</a></td>
    <td>Fast LD calculation from VCFs by exploiting sparsity and haplotype structure.</td>
  </tr>
  <tr>
    <td nowrap><a href="https://github.com/pfultz2/cget">cget</a></td>
    <td nowrap><a href="https://github.com/pfultz2/cget/commits?author=welchr">Commits</a> • <a href="https://github.com/pfultz2/cget/pulls?q=is%3Apr+author%3Awelchr+">PRs</a></td>
    <td>C++ package retrieval for CMake build systems.</td>
  </tr>
</tbody>
</table>

## Archived

Projects no longer under development:

<table>
<thead>
  <tr>
    <th align="left" width="195px">Repo</th>
    <th align="left" width="125px">Contributions</th>
    <th align="left">Description</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td nowrap><a href="https://github.com/statgen/swiss">swiss</a></td>
    <td nowrap><a href="https://github.com/statgen/swiss/commits?author=welchr">Commits</a> • <a href="https://github.com/statgen/swiss/pulls?q=is%3Apr+author%3Awelchr+">PRs</a></td>
    <td>Software to find overlaps between significant GWAS association results and a GWAS catalog of known genetic associations using LD or genomic distance.</td>
  </tr>
  <tr>
    <td nowrap><a href="https://github.com/statgen/locuszoom-standalone">locuszoom-standalone</a></td>
    <td nowrap><a href="https://github.com/statgen/locuszoom-standalone/commits?author=welchr">Commits</a> • <a href="https://github.com/statgen/locuszoom-standalone/pulls?q=is%3Apr+author%3Awelchr+">PRs</a></td>
    <td>Command line version of the original <a href="https://genome.sph.umich.edu/wiki/LocusZoom_Standalone">LocusZoom</a> (circa 2010). This has been deprecated in favor of the new interactive <a href="https://github.com/statgen/locuszoom">LocusZoom.js</a>. Please try it out on the new <a href="https://my.locuszoom.org/">LocusZoom website</a>.</td>
  </tr>
  <tr>
    <td nowrap><a href="https://github.com/welchr/mvtnorm.js">mvtnorm.js</a></td>
    <td nowrap></td>
    <td>A test of cross-compiling the multivariate normal distribution function into JS/WASM from Fortran.</td>
  </tr>
</tbody>
</table>

# Tutorials/Talks

[Examples of submitting SLURM jobs](https://github.com/welchr/SLURM-examples) on our cluster.

[Tutorial](https://github.com/welchr/csg-snakemake) on using Snakemake.


[LocusZoom.js]: https://github.com/statgen/locuszoom
[raremetal.js]: https://github.com/statgen/raremetal.js
[**raremetal.js**]: https://github.com/statgen/raremetal.js
