
# About

I am currently a bioinformatician and software developer at the [Center for Statistical Genetics](https://sph.umich.edu/csg/), University of Michigan.

# Projects

## Active

[**LDServer**](https://github.com/statgen/LDServer). Serves linkage disequilibrium (LD) data over a REST API. Used to calculate LD on-the-fly for the new interactive [LocusZoom.js] viewer, and score statistics/covariance matrices for use with [raremetal.js].

[**raremetal.js**]. JS package for calculating rare variant aggregation tests. Can calculate in the browser or server-side with node.js. Supports burden, VT, SKAT, and SKAT-O.

[**LocusZoom API**](https://github.com/statgen/locuszoom-api). Backend REST API service for [LocusZoom.js]. Serves information on genes, recombination rates, EBI GWAS catalog variants, genomic interval annotations, dbSNP metadata, and more.

## Maintenance only

[**swiss**](https://github.com/statgen/swiss). Software to find overlaps between significant GWAS association results and a GWAS catalog of known genetic associations using LD or genomic distance.

## Unsupported

[**locuszoom-standalone**](https://github.com/statgen/locuszoom-standalone). Command line version of the original [LocusZoom](https://genome.sph.umich.edu/wiki/LocusZoom_Standalone) (circa 2010). This has been deprecated in favor of the new interactive [LocusZoom.js]. Please try it out on the new [LocusZoom website](https://my.locuszoom.org/).

[**mvtnorm.js**](https://github.com/welchr/mvtnorm.js). A test of cross-compiling the multivariate normal distribution function into JS/WASM from Fortran.

# Tutorials/Talks

[Examples of submitting SLURM jobs](https://github.com/welchr/SLURM-examples) on our cluster.

[Tutorial](https://github.com/welchr/csg-snakemake) on using Snakemake.


[LocusZoom.js]: https://github.com/statgen/locuszoom
[raremetal.js]: https://github.com/statgen/raremetal.js
[**raremetal.js**]: https://github.com/statgen/raremetal.js
